### PR TITLE
Make node-fallbacks build as esm not cjs / Fix `node-fetch`

### LINKED ISSUE #3303: fetch is not a function. It's Object.
### What version of Bun is running?

0.6.9

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Errors while running application with [Node-Fetch](https://www.npmjs.com/package/node-fetch) as a dependency within a 3rd party package [bfx-api-node-rest](https://github.com/bitfinexcom/bfx-api-node-rest).

A test code:

````node
import {type Serve} from "bun";

const { RESTv2 } = require('bfx-api-node-rest');

const rest = new RESTv2({
  apiKey: 'asdf',
  apiSecret: 'asdf',
  transform: true, // to have full models returned by all methods
});

export default {
  fetch: async (resp) => {
    const response = await rest.wallets({});

    console.log(response)

    return new Response('test')
  }
} satisfies Serve;
````

### What is the expected behavior?

Connect to the Bitfinex API and return he list of crypto wallets.

### What do you see instead?

The following error when I open `http://localhost:3000`:

````node
139 |     return !!this._agent
140 |   }
141 |
142 |   async _request (url, reqOpts, transformer, cb) {
143 |     try {
144 |       const resp = await fetch(url, reqOpts)
                             ^
TypeError: fetch is not a function. (In 'fetch(url, reqOpts)', 'fetch' is an instance of Object)
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:144:25
      at _request (/Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:142:18)
      at _makeAuthRequest (/Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:180:26)
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/index.ts:13:27
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/index.ts:12:16
GET - http://localhost:3000/ failed
139 |     return !!this._agent
140 |   }
141 |
142 |   async _request (url, reqOpts, transformer, cb) {
143 |     try {
144 |       const resp = await fetch(url, reqOpts)
                             ^
TypeError: fetch is not a function. (In 'fetch(url, reqOpts)', 'fetch' is an instance of Object)
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:144:25
      at _request (/Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:142:18)
      at _makeAuthRequest (/Users/mariodian/Source/Bitcoin/bfx-tools/src/server/node_modules/bfx-api-node-rest/lib/rest2.js:180:26)
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/index.ts:13:27
      at /Users/mariodian/Source/Bitcoin/bfx-tools/src/server/index.ts:12:16
````

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by paperclover**
Fixes #3303

The CommonJS doesnt work if the module itself is a CommonJS module, etc

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
we can probably just use bun's bundler now

**Comment by Jarred-Sumner** [comment]
@paperdave looks like this breaks the "querystring" package:

image

**Comment by Jarred-Sumner** [comment]
@paperdave for each case with an `export * from`, we also need an `export default`

**Comment by paperclover** [comment]
ended up going through every single file and triple checking it functioned properly.

all the modules are esm now and we can move to using bun to bundle these later.

File: test/js/node/fs/node-fetch.test.js [hunk 1]
```diff
@@ -19,3 +19,15 @@ test("node-fetch fetches", async () => {
...
+
+test("node-fetch.default fetches", async () => {
+  const server = Bun.serve({
+    port: 0,
+    fetch(req, server) {
+      server.stop();
+      return new Response();
+    },
+  });
+  expect(await fetch2("http://" + server.hostname + ":" + server.port)).toBeInstanceOf(Response);
```
Comment (Jarred-Sumner) [inline test/js/node/fs/node-fetch.test.js:31]: nice

File: src/node-fallbacks/crypto.js [hunk 2]
```diff
@@ -79,3 +80,13 @@ if (timingSafeEqual) {
 }
 
 export const webcrypto = crypto;
+
+export default {
+  ...cryptoBrowserify,
+  getRandomValues,
+  randomUUID,
+  timingSafeEqual,
```
Comment (Jarred-Sumner) [inline src/node-fallbacks/crypto.js:88]: Some of these don't make much sense anymore, but we can address that in a follow-up PR

This was originally used in bun's runtime as well. Until we started added more stuff to it.

**Comment by Jarred-Sumner** [comment]
nice

**Pull request merged**